rs/tests/nested/src/lib.rs

```diff
@@ -101,32 +91,27 @@ pub fn registration(env: TestEnv) {
 
 /// Upgrade each HostOS VM to the test version, and verify that each is
 /// healthy before and after the upgrade.
-pub fn upgrade(env: TestEnv, to: &NestedVersionTarget) {
+pub fn upgrade(env: TestEnv) {
     let logger = env.logger();
 
-    let (target_version, url, sha256) = match to {
-        NestedVersionTarget::Mainnet => (
-            env.get_mainnet_version().unwrap(),
-            env.get_mainnet_hostos_update_img_url().unwrap(),
-            env.get_mainnet_hostos_update_img_sha256().unwrap(),
-        ),
-        NestedVersionTarget::Branch(false) => (
-            env.get_branch_version().unwrap(),
-            env.get_hostos_update_img_url().unwrap(),
-            env.get_hostos_update_img_sha256().unwrap(),
-        ),
-        NestedVersionTarget::Branch(true) => (
-            format!("{}-test", env.get_branch_version().unwrap()),
-            env.get_hostos_update_img_test_url().unwrap(),
-            env.get_hostos_update_img_test_sha256().unwrap(),
-        ),
-        NestedVersionTarget::Published {
-            version,
-            url,
-            sha256,
-        } => (version.to_owned(), url.to_owned(), sha256.to_owned()),
-    };
-    let target_version = HostosVersion::try_from(target_version).unwrap();
+    let target_version = HostosVersion::try_from(
+        read_dependency_from_env_to_string(HOSTOS_UPDATE_VERSION_ENV_VAR).unwrap_or_else(|_| {
+            eprintln!("environment variable {HOSTOS_UPDATE_VERSION_ENV_VAR} is not set");
+            process::exit(1);
```

Comment:
Same here and below. Instead of eprintln + exit, you could panic and allow destructors to run.